code/controllers/subsystems/initialization/persistence.dm

```diff
@@ -30,9 +32,32 @@ SUBSYSTEM_DEF(persistence)
 		P.Shutdown()
 
 	// Refer to snowflake above.
-	if(fexists(elevator_fall_path))
-		fdel(elevator_fall_path)
-	text2file("[elevator_fall_shifts]", elevator_fall_path)
+	if(fexists(ELEVATOR_FALL_PATH))
+		fdel(ELEVATOR_FALL_PATH)
+	text2file("[elevator_fall_shifts]", ELEVATOR_FALL_PATH)
+
+	// Handle level data shutdown.
+	start_persistent_level_save()
+	while(_persistent_save_running)
+		sleep(1)
+
+/datum/controller/subsystem/persistence/fire(resumed)
+	start_persistent_level_save()
+
+/datum/controller/subsystem/persistence/proc/start_persistent_level_save()
+	// Is this actually desirable? People moving around or modifying
+	// atoms across save could result in inconsistent data.
+	set waitfor = FALSE
```

Comment:
i think you'd have to actually add `if(_persistent_save_running) return` hooks into move and other client-driven/verb things to make it block anyway, is the thing. i'm not sure `waitfor = TRUE` would be enough